$OpenBSD: patch-include_private_gcconfig_h,v 1.20 2015/06/27 13:21:29 jasper Exp $

- All OpenBSD mips platforms are 64-bit, so correct the wordsize and alignment
- use __data_start instead of _fdata on OpenBSD/mips64

--- include/private/gcconfig.h.orig	Sat Jun 27 13:48:41 2015
+++ include/private/gcconfig.h	Sat Jun 27 13:51:16 2015
@@ -1635,14 +1635,15 @@
 #  endif
 #  ifdef OPENBSD
 #    define OS_TYPE "OPENBSD"
-#    define ALIGNMENT 4
+#     define CPP_WORDSZ 64
+#     define ALIGNMENT 8
 #     ifndef GC_OPENBSD_THREADS
 #      include <sys/param.h>
 #      include <uvm/uvm_extern.h>
 #      define STACKBOTTOM ((ptr_t) USRSTACK)
 #    endif
-     extern int _fdata[];
-#    define DATASTART ((ptr_t)_fdata)
+     extern int __data_start[];
+#    define DATASTART ((ptr_t)__data_start)
      extern char _end[];
 #    define DATAEND ((ptr_t)(&_end))
 #    define DYNAMIC_LOADING
